server/environment/environment.go

```diff
@@ -66,6 +66,8 @@ type Env interface {
 	GetActionCacheClient() repb.ActionCacheClient
 	GetByteStreamClient() bspb.ByteStreamClient
 	SetByteStreamClient(bspb.ByteStreamClient)
+	GetGrpcClientConnPoolCache() interfaces.GrpcClientConnPoolCache
```

Comment:
Does this need to be an app wide service? It seems simpler to me to have the localized use cases instantiate their own cache as I don't expect this to be a very common pattern. At this time it's only going to be used in a single place. 

cacheproxy.go has a similar implementation but it has some quirks that I think that would make it hard to reuse the same code for both.

Reply:
I had been kidding myself that maybe we should share this with other code that's dialing (i.e., share pools between all of that code as well), but the cacheproxy code makes me... not think so.

Anyway, this was mostly out of expedience (this approach made me not have to convert the byte stream client to an object so that it could hold a reference to the cache).  I was already planning to convert it to an object, it's just a matter of if I do it now or later.